arduino/smartcar/smartcar.ino

```diff
@@ -220,11 +254,43 @@ bool reactToSensor(int sensorDistance, int STOP_DISTANCE){
     }
     return false;
 }
+
+/**
+ * @param sensor : takes the name of the sensor to be checked  as a String.
+ * @Returns true if sensor does not detect an obstacle or if it is too close to be detected. False if an obstacle is detected within the range of the sensor.
+ */
+bool isClear(String sensor)
+{
+    if (sensor == "frontUS"){
+        return (frontUS.getDistance() == NO_OBSTACLE_VALUE);
+    } else if(sensor == "frontIR") {
+        return (frontIR.getDistance() == NO_OBSTACLE_VALUE);
+    }else if(sensor == "backIR"){
+        return (backIR.getDistance() == NO_OBSTACLE_VALUE);
+    }else if(sensor == "rightIR"){
+        return (rightIR.getDistance() == NO_OBSTACLE_VALUE);
+    }else if(sensor == "leftIR"){
+        return (leftIR.getDistance() == NO_OBSTACLE_VALUE);
+    }else{
+        return false;
+    }
+}
+
+/**
+ * Slows down the car smoothly by dividing the speed by 3 until it reaches a safe speed to stop.
+ */
+void slowDownSmoothly()
+{
+    Serial.println("Slowing down");
+    if (car.getSpeed() >= STOPPING_SPEED){//check constant for details
+        car.setSpeed(convertSpeed(car.getSpeed()) * 0.7);
+    }
+}
+
 /**
  * Method to look and compare two sensor values (100ms apart) to know if an obstacle is coming towards the car
  */
 void reactToSides() {
```

Comment:
in the reactToSides(), there are possible data loss from int getDistance() to float.